kernel/Kconfig

```diff
@@ -928,4 +928,16 @@ config THREAD_LOCAL_STORAGE
 
 endmenu
 
+menu "Device Options"
+
+config HAS_DYNAMIC_DEVICE_HANDLES
+	bool
+	depends on (PM_DEVICE_POWER_DOMAIN_DYNAMIC_NUM > 0)
+	default y
```

Comment:
hidden options should be selected from somewhere else, we're adding here PM subsys dependencies into `kernel/Kconfig`.

Reply:
ok, that is fair. It will necessary an additional symbol to select this one I believe though